### PR TITLE
Initial tests for process tree inspection tools

### PR DESCRIPTION
**Pull request by JayH5**

### CODE REVIEW

File: seaworthy/ps.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    """
+    Build a tree structure from a list of PsRow objects.
+    :param ps_rows: a list of PsRow objects
+    :return: a PsTree object
+    """
+    ps_tree = None
+    for row in ps_rows:
+        if row.ppid == '0':
+            if ps_tree is not None:
+                raise PsException("Too many init processes (ppid=0) found")
```
Comment (JayH5) [inline seaworthy/ps.py]: I wouldn't call these init processes, rather PID 1 processes or something. An init system is optional.

File: seaworthy/tests/test_ps.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
+import pytest
+
+from seaworthy.ps import PsException, PsRow, PsTree, build_process_tree
+
+
+def mkrow(pid, ppid, ruser='root', args=None):
+    if args is None:
```
Comment (JayH5) [inline seaworthy/tests/test_ps.py]: Coverage is saying this branch is not being exercised.

File: seaworthy/ps.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    """
+    Build a tree structure from a list of PsRow objects.
+    :param ps_rows: a list of PsRow objects
+    :return: a PsTree object
+    """
+    ps_tree = None
+    for row in ps_rows:
+        if row.ppid == 0:
+            if ps_tree is not None:
+                raise PsException("Too many process tree roots (ppid=0) found")
```
Comment (JayH5) [inline seaworthy/ps.py]: I've been preferring `'` to `"`.. but I don't know if there's a single Python project that is consistent in that regard.

Comment (jerith) [inline seaworthy/ps.py]: The ones I've seen usually prefer `"` to `'`. I'm not very consistent. :-/

**Pull request merged**